### PR TITLE
[MM-31497] - Add spec for cloud/subscription/stats

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds a new cloud only endpoint to obtain stats about the kind of subscription a user has.

#### Ticket Link
[MM-31497](https://mattermost.atlassian.net/browse/MM-31497?atlOrigin=eyJpIjoiMDJhMDQwOWM1ODZhNDMzM2IyNDE5ZjAxNDM1OTg4Y2UiLCJwIjoiaiJ9)

#### Related Pull Request
[https://github.com/mattermost/mattermost-server/pull/16786](https://github.com/mattermost/mattermost-server/pull/16786)

### CODE REVIEW

File: v4/source/cloud.yaml [hunk 1]
```diff
@@ -319,3 +319,26 @@
...
+        - cloud
+      summary: GET endpoint for cloud subscription stats
+      description: >
+        An endpoint for that returns stats about a users subscription. For example remaining seats on a free tier
+
+        ##### Permissions
+
+        This endpoint should only be accessed in a Mattermost Cloud instance
+
+        __Minimum server version__: 5.30
```
Comment (ethervoid) [inline v4/source/cloud.yaml:334]:
```suggestion
        __Minimum server version__: 5.34
```

File: v4/source/definitions.yaml [hunk 2]
```diff
@@ -2641,6 +2641,14 @@ components:
           type: integer
         dns:
           type: string
+    SubscriptionStats:
+      type: object
+      properties:
+        remaining_seats:
+          type: integer
+          format: int
```
Comment (ethervoid) [inline v4/source/definitions.yaml:2649]: I don't think the format option is needed here

File: v4/source/cloud.yaml [hunk 3]
```diff
@@ -319,3 +319,26 @@
...
+        __Note:__ This is intended for internal use and is subject to change.
+      responses:
```
Comment (ethervoid) [inline v4/source/cloud.yaml:336]: What about 401/403 responses if the permissions are not right?

File: v4/source/cloud.yaml [hunk 4]
```diff
@@ -319,3 +319,26 @@
           $ref: "#/components/responses/Forbidden"
         "501":
           $ref: "#/components/responses/NotImplemented"
+  /cloud/subscription/stats:
+    get:
+      tags:
+        - cloud
+      summary: GET endpoint for cloud subscription stats
+      description: >
+        An endpoint for that returns stats about a users subscription. For example remaining seats on a free tier
```
Comment (ethervoid) [inline v4/source/cloud.yaml:328]:
```suggestion
        An endpoint that returns stats about a user's subscription. For example remaining seats on a free tier
```

File: v4/source/cloud.yaml (hunk 3 above)
Comment (AGMETEOR) [reply v4/source/cloud.yaml:336]: Hello @ethervoid , thanks a lot for the review. This endpoint doesn't require any permissions.

**Review by ethervoid — APPROVED** [review]
LGTM! Good work!

**Pull request merged**